### ISSUE TITLE
Optimised images are saving upside down

### ISSUE THREAD

**Issue by harrync** [pornel/ImageOptim#82]
After processing some images, I've noticed that some have been rotated 180deg so are upside down. It doesn't seem to happen with every single image / specific filetype but is more common than not.

I'm on late 2013 retina MBP 2.3i7 16GB on Yosemite

Cheers
Harry

**Comment by mathiasbynens** [comment]
Sounds like EXIF orientation meta data getting stripped. This is to be expected.

**Comment by pornel** [comment]
Yes, indeed. An ideal solution would be to parse EXIF an do lossless transformation to rotate actual data.

Does anybody know a minimal EXIF parser? ExifTool is great, but it supports so much that it alone is 2 times bigger than all ImageOptim tools.

**Comment by Jellyfrog** [comment]
>Does anybody know a minimal EXIF parser? ExifTool is great, but it supports so much that it alone is 2 times bigger than all ImageOptim tools combined.

What about Apple API?
https://developer.apple.com/library/mac/documentation/GraphicsImaging/Reference/CGImageSource/index.html#//apple_ref/doc/uid/TP40001399
https://developer.apple.com/library/mac/documentation/GraphicsImaging/Reference/CGImageProperties_Reference/index.html#//apple_ref/doc/constant_group/EXIF_Dictionary_Keys

**Comment by ace-dent** [comment]
I use jhead for this: http://www.sentex.net/~mwandel/jhead/usage.html
and something like - `jhead -q -autorot %1`.
You can also strip out thumbnails if you want to save a few more bytes... `jhead -purejpg -dt %1` .

**Comment by pornel** [comment]
Great. Thanks for suggestions.

**Comment by harrync** [comment]
Thanks for responses all, will give these a try
